src/Native/Runtime/arm/InteropThunksHelpers.S

```diff
@@ -2,50 +2,55 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-#include <unixasmmacros.inc>
-
 .syntax unified
 .thumb
 
-#ifdef _DEBUG
-.rodata
-AssertMsg:                              .asciz "__FILE__:%s: %s is not implemented\n"
-FileName:                               .asciz "InteropThunksHelpers.S"
-RhCommonStubName:                       .asciz "RhCommonStub"
-RhGetCommonStubAddressName:             .asciz "RhGetCommonStubAddress"
-RhGetCurrentThunkContextName:           .asciz "RhGetCurrentThunkContext"
-
-.text
-.macro GEN_ASSERT_FUNC func
-        GEN_ASSERT AssertMsg, FileName, \func
-.endm
-#endif
-
-// TODO: Implement Arm support
+#include <unixasmmacros.inc>
+#include <AsmOffsets.inc>         // generated by the build from AsmOffsets.cpp
+#define POINTER_SIZE 4
 
 //
 // RhCommonStub
 //
 NESTED_ENTRY RhCommonStub, _TEXT, NoHandler
-#ifdef _DEBUG
-        GEN_ASSERT_FUNC RhCommonStubName
-#endif
+          // Custom calling convention:
+          //      red zone has pointer to the current thunk's data block (data contains 2 pointer values: context + target pointers)
+          //      Copy red zone value into r12 so that the PROLOG_PUSH doesn't destroy it
+          ldr          r12, [sp, #-4]
+          PROLOG_PUSH  "{r0-r5}"
```

Comment:
I think that the TLS helper called by `INLINE_GET_TLS_VAR` may corrupt floating point arg registers. You may need to save floating point arg registers here (similar to what is done in https://github.com/dotnet/corert/blob/master/src/Native/Runtime/amd64/InteropThunksHelpers.S#L16).